### ISSUE TITLE
[Question] Runtime vs. number of ROIs

### ISSUE THREAD

**Issue by HaixinLiuNeuro** [MouseLand/cellpose#734]
First, I'm grateful to have this great tool to use! Thank you, cellpose team!

I was wondering about how number of ROIs affect the time used for segmenting a image. 

I tried some images:
![image](https://github.com/MouseLand/cellpose/assets/46263698/2b0d3278-3466-4768-8281-5698df767b12)
x axis is detected ROI number, y axis is time used in seconds. All images are ~ 2k x 2k. Color indicates different markers expressed. 

It is somewhat linear in general. 

I'm not sure how to compare/understand with the benchmarking result: 
![image](https://github.com/MouseLand/cellpose/assets/46263698/a495b214-83f8-4827-a8af-982425674f42)

My specific questions are:
1a. Is the increase of time used in my test mostly due to the postprocessing step? 
1b. if so, anyway to speed up the process?
2. Since all my images are the same size, will making an image stack to feed into cellpose faster than running each image sequentially?

Thank you!

**Comment by mrariden** [comment]
1a. Yes, the neural network itself doesn't keep track of the number of objects
1b. You can use a GPU to speed up the postprocessing. It's unlikely that we will significantly speed the post processing algorithm up. A lot of optimization when into the current algorithm. It's using pytorch to leverage GPU acceleration to do the reconstruction from the IR net output. The number of flow following iterations is set to 200, which you could attempt to reduce to speed up the reconstruction. Result quality will depend on your data/models
2. Maybe, you could try an experiment and find out. If you're using a GPU my guess would be yes.

**Comment by HaixinLiuNeuro** [comment]
Thank you Michael @mrariden ! 

1b, the current postprocessing step is done on CPU, correct? probably naive question, How can I opt to use GPU? Is there some option built in cellpose (I'm not good at playing with GPU computation, sorry)? 
2. Yes. I am testing with my laptop GPU, not great, but OK to play (thanks to cellpose relative small model comparing to other models) (GPU info: NVIDIA Quadro T1000 4 GB memory). 
I tried with some my testing data:
20 2k x 2k images with ROI number ranging from 100-600 (run in jupyter notebook)
The time spent to return the masks are very similar ~ 210 seconds. 
I also varied batch size from 8, 16, 24. It did not change the run time dramatically. 

3. I also tried to scale my images (1-99 percentile method as cellpose's default, and set normalize = false) before feeding into cellpose. The run time is similar (probably I gained 10 seconds in total) 

Do those sounds normal to you? (or I was not doing it correctly) 

Thank you!

**Comment by mrariden** [comment]
If you use the GPU option either in the CLI or the API, it will do the post processing on the GPU. If you had this turned off before, it would not use the GPU. Note that not all GPUs are the same, and generally more memory is better.

Those results sound reasonable. 

I'm closing this for now, reopen if you have other issues.
